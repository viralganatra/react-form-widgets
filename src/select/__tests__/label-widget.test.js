import React from 'react';
import LabelWidget from '../label-widget';

function noop() {}

const props = {
    label: 'test',
    placeholder: 'test',
    className: 'bar',
    onClick: noop,
};

describe('LabelWidget', () => {
    it('should render correctly', () => {
        const wrapperNoLabel = shallow(
            <LabelWidget {...props} label={null} />,
        );

        const wrapperLabel = shallow(
            <LabelWidget {...props} />,
        );

        expect(wrapperNoLabel).toMatchSnapshot();
        expect(wrapperLabel).toMatchSnapshot();
    });

    it('should validate the prop types', () => {
        const wrapperInvalidLabel = () => shallow(
            <LabelWidget {...props} label={2} />,
        );

        const wrapperInvalidPlaceholder = () => shallow(
            <LabelWidget {...props} placeholder={2} />,
        );

        const wrapperInvalidOnClick = () => shallow(
            <LabelWidget {...props} onClick={[]} />,
        );

        expect(wrapperInvalidLabel).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidPlaceholder).toThrowErrorMatchingSnapshot();
        expect(wrapperInvalidOnClick).toThrowErrorMatchingSnapshot();
    });
});
