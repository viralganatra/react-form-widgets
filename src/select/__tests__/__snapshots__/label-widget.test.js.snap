exports[`LabelWidget should render correctly 1`] = `
<div
  className="bar"
  onClick={[Function]}
  style={
    Object {
      "color": "grey",
    }
  }>
  test
</div>
`;

exports[`LabelWidget should render correctly 2`] = `
<div
  className="bar"
  onClick={[Function]}>
  test
</div>
`;

exports[`LabelWidget should validate the prop types 1`] = `
"Warning: Failed prop type: Invalid prop \`label\` of type \`number\` supplied to \`SelectLabel\`, expected \`string\`.
    in SelectLabel"
`;

exports[`LabelWidget should validate the prop types 2`] = `
"Warning: Failed prop type: Invalid prop \`placeholder\` of type \`number\` supplied to \`SelectLabel\`, expected \`string\`.
    in SelectLabel"
`;

exports[`LabelWidget should validate the prop types 3`] = `
"Warning: Failed prop type: Invalid prop \`onClick\` of type \`array\` supplied to \`SelectLabel\`, expected \`function\`.
    in SelectLabel"
`;
